### PR TITLE
TB specific changes to Safe Inputs

### PR DESCRIPTION
**Pull request by niranjan-ramesh**
TB specific enhancements for safe inputs.

- Added validation of excel input
- Test schema used - schema yet to be finalized for PTs

### CODE REVIEW

**Comment by Collinbrown95** [comment]
@niranjan-ramesh I think in the spirit of "1 PR = 1 feature", the only thing I'd like to flesh out in this PR is a high-level overview of what the user experience for data validation looks like. We can iterate on this, of course, but I think getting a minimum implementation of editable tables + tooltips is a good starting point.

If you're up for it, I propose we implement some of the UX changes discussed in the review thread, then merge this into the `main` branch and tackle next steps in a separate branch.

Great work, this is an awesome feature! :)

**Comment by niranjan-ramesh** [comment]
@Collinbrown95 Thank you for checking the branch. That sounds great. We can get started on these changes.

**Review by Collinbrown95 — CHANGES_REQUESTED** [review]
Finishing the review.

File: ui/src/components/PaginatedTable.tsx
[the remark this answers is not in the document]
Comment (niranjan-ramesh) [reply ui/src/components/PaginatedTable.tsx:1]: @Collinbrown95 I have pushed changes implementing (1) and (2) and have made some re-organization including (3). 

image

image

EditableCell is a component with the data, error label and callbacks to modify the original data and re-run validation. 

I would love to get your feedback on the changes.

File: ui/src/utils/errorUtils.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    if (Array.isArray(value.type)) return value.type[0]
+    return value.type
+})
+
+export const headers: {header: string; type: string}[] = Object.keys(schema.properties).map((header, index) => ({
+    header,
+    type: fieldTypes[index]
+}))
+
+export const constructErrorMessage = (error: ErrorObject):string => {
```
[the remark this answers is not in the document]
Comment (Collinbrown95) [reply ui/src/utils/errorUtils.ts:32]: As discussed offline, we'll incorporate this refactor into a separate PR later on.

**Review by Collinbrown95 — APPROVED** [review]
Looks good to me!

**Pull request merged**